frontend/public/components/modals/roll-back-rc-modal.jsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
+import { withHandlePromise } from '../utils';
+import { DeploymentConfigModel } from '../../models';
+import { k8sCreate, k8sUpdate } from '../../module/k8s';
+
+const BaseRollBackRCModal = withHandlePromise((props) => {
+  const { t } = useTranslation();
+  const dcName = props.resource.metadata.ownerReferences[0].name;
```

Comment:
I'd use optional chaining here just in case and handle the scenario where the DC name is not set.

To be strictly correct, we should find the owner reference to the correct `kind` in case there are multiple owner references. Do have a utility for this logic somewhere used by the topology view?